tools/ob_admin/backup_tool/ob_admin_backup_validation_task.h

```diff
@@ -0,0 +1,606 @@
+ * Copyright (c) 2024 OceanBase
+ * OceanBase CE is licensed under Mulan PubL v2.
+ * You can use this software according to the terms and conditions of the Mulan
+ * PubL v2. You may obtain a copy of Mulan PubL v2 at:
+ *          http://license.coscl.org.cn/MulanPubL-2.0
+ * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY
+ * KIND, EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+ * NON-INFRINGEMENT, MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE. See the
+ * Mulan PubL v2 for more details.
+ */
+
+#ifndef OB_ADMIN_BACKUP_VALIDATION_TASK_H_
+#define OB_ADMIN_BACKUP_VALIDATION_TASK_H_
+#include "../ob_admin_executor.h"
+#include "logservice/archiveservice/large_buffer_pool.h"
+#include "logservice/ob_log_external_storage_handler.h"
+#include "ob_admin_backup_validation_ctx.h"
+
+namespace oceanbase
+{
+namespace tools
+{
+
+// Dagnet
+struct ObAdminDataBackupValidationDagInitParam final : public ObIDagInitParam
+{
+  ObAdminDataBackupValidationDagInitParam() {}
+  ObAdminDataBackupValidationDagInitParam(ObAdminBackupValidationCtx *ctx) : ctx_(ctx) {}
+  virtual ~ObAdminDataBackupValidationDagInitParam() {}
+  virtual bool is_valid() const override { return true; }
+  ObAdminBackupValidationCtx *ctx_;
+};
+class ObAdminDataBackupValidationDagNet final : public share::ObIDagNet
+{
+public:
+  ObAdminDataBackupValidationDagNet()
+      : share::ObIDagNet(ObDagNetType::DAG_NET_DATA_BACKUP_VALIDATION), is_inited_(false), id_(0),
+        ctx_(nullptr)
+  {
+  }
+  virtual ~ObAdminDataBackupValidationDagNet();
+  int init(ObAdminBackupValidationCtx *ctx);
+  int init_by_param(const ObIDagInitParam *param);
+  virtual bool is_valid() const override;
+  virtual int start_running() override;
+  virtual bool operator==(const share::ObIDagNet &other) const override;
+  virtual int fill_dag_net_key(char *buf, const int64_t buf_len) const override;
+  virtual int fill_comment(char *buf, const int64_t buf_len) const override;
+  virtual int64_t hash() const override;
+  virtual bool is_ha_dag_net() const override { return false; }
+  INHERIT_TO_STRING_KV("ObIDagNet", share::ObIDagNet, K_(is_inited));
+
+private:
+  bool is_inited_;
+  int64_t id_;
+  ObAdminBackupValidationCtx *ctx_;
+  DISALLOW_COPY_AND_ASSIGN(ObAdminDataBackupValidationDagNet);
+};
+struct ObAdminLogArchiveValidationDagInitParam final : public ObIDagInitParam
+{
+  ObAdminLogArchiveValidationDagInitParam() {}
+  ObAdminLogArchiveValidationDagInitParam(ObAdminBackupValidationCtx *ctx) : ctx_(ctx) {}
+  virtual ~ObAdminLogArchiveValidationDagInitParam() {}
+  virtual bool is_valid() const override { return true; }
+  ObAdminBackupValidationCtx *ctx_;
+};
+class ObAdminLogArchiveValidationDagNet final : public share::ObIDagNet
+{
+public:
+  ObAdminLogArchiveValidationDagNet()
+      : share::ObIDagNet(ObDagNetType::DAG_NET_LOG_ARCHIVE_VALIDATION), is_inited_(false), id_(0),
+        ctx_(nullptr)
+  {
+  }
+  virtual ~ObAdminLogArchiveValidationDagNet();
```

Comment:
if use final do not need to use virtual here

Reply:
all fixed